### PR TITLE
Fix check if cookbook is already uploaded (bsc#980888)

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

File: crowbar_framework/lib/crowbar/chef/upload.rb [hunk 1]
```diff
@@ -227,15 +227,12 @@ def local_cookbook_md5_checksums(cookbook)
...
-        local_md5sums = local_cookbook_md5_checksums(cookbook)
-        remote_md5sums = api_cookbook_md5_checksums(cookbook.name)
-        combined_md5sums = local_md5sums.deep_merge(remote_md5sums)
-
-        combined_md5sums.reject! { |cb, _| local_md5sums.include? cb }
-        return false unless combined_md5sums.empty?
-
-        logger.info("Cookbook #{cookbook.name} already exists, skipping...")
-        true
+        if local_cookbook_md5_checksums(cookbook) == api_cookbook_md5_checksums(cookbook.name)
```
Comment (nicolasbock) [inline crowbar_framework/lib/crowbar/chef/upload.rb]: Why is this so much simpler now? I mean, did something change in what `*_cookbook_md5_checksums()` return?

Comment (rsalevsky) [inline crowbar_framework/lib/crowbar/chef/upload.rb]: Nop, but when I wrote the code I tough way too complex. I did forgot that it's possible to compare hashs in ruby without caring about the order. I tried to implement something which checks if a hash is the same but `deep_merge` did not work as expected so at the end it was always true.

Comment (nicolasbock) [inline crowbar_framework/lib/crowbar/chef/upload.rb]: Ah, this looks way better now :smile:

**Comment by nicolasbock** [comment]
+1

**Comment by nicolasbock** [comment]
Gating is failing. Could you have a look?

**Comment by jsuchome** [comment]
+1

**Comment by MaximilianMeister** [comment]
+1

**Pull request merged**